orb-ui/cone/src/led.rs

```diff
@@ -0,0 +1,140 @@
+use color_eyre::eyre;
+use color_eyre::eyre::Context;
+use ftdi_embedded_hal::eh1::spi::SpiBus;
+use ftdi_embedded_hal::libftd2xx::{Ft4232h, Ftdi, FtdiCommon};
+use orb_rgb::Argb;
+use std::sync::{Arc, Mutex};
+use tokio::sync::{mpsc, watch};
+use tokio::task;
+
+pub struct Led {
+    spi: ftdi_embedded_hal::Spi<Ft4232h>,
```

Comment:
`spi` should be owned by the task, not by the proxy struct. The point of the task is to provide a single spot that has exclusive access to the hardware, and the proxy object communicates with it and manages the task's lifetime (and optionally provides RPC or a way to get a channel to communicate with it).

Most probably, if you move `spi` into the task, you will no longer need the Arc<Mutex>.